### PR TITLE
Implement loader as a global in runtime, as a temporary hack.

### PR DESCRIPTION
**Pull request by raulverag**

### CODE REVIEW

File: src/runtime/runtime.ts [hunk 1]
```diff
@@ -54,4 +44,17 @@ export class Runtime {
...
+  // TODO(raulverag): Move this stuff to ArcRunner, when it exists, as it will
+  // be part of the per-persona environment.
+  static _loader: Loader;
+
+  static setLoader(loader: Loader) {
+    Runtime._loader = loader;
+  }
+
+  static getLoader(): Loader {
+    return Runtime._loader;
```
Comment (shans) [inline src/runtime/runtime.ts]: assert that the loader exists?

Comment (raulverag) [reply src/runtime/runtime.ts]: Done.

**Comment by sjmiles** [comment]
I need to understand the impetus for this PR before I can review it properly. Is there a corresponding issue?

File: shells/lib/env/env-base.js [hunk 2]
```diff
@@ -1,4 +1,5 @@
 import {Env} from '../arcs.js';
+import {Runtime} from '../../../build/runtime/runtime.js';
```
Comment (sjmiles) [inline shells/lib/env/env-base.js]: Why is env-base being changed at all?

File: shells/lib/env/env-base.js [hunk 3]
```diff
@@ -28,9 +32,8 @@ export class EnvBase {
       'https://$artifacts/': `${root}/particles/`, // deprecated
     };
   }
-  // loader construction is lazy, so pathMap can be configured prior
```
Comment (sjmiles) [inline shells/lib/env/env-base.js]: Laziness was part of the contract of this object, we can't just eliminate it without taking care.

File: shells/lib/env/env-base.js [hunk 4]
```diff
@@ -9,6 +10,9 @@ export class EnvBase {
     this.rootPath = rootPath;
     this.loaderKind = loaderKind;
     this.pathMap = this.createPathMap(rootPath);
+    this._loader = new (this.loaderKind)(this.pathMap);
+    // TODO. Temporary until ArcRunner carries this info.
+    Runtime.setLoader(this._loader);
```
Comment (sjmiles) [inline shells/lib/env/env-base.js]: If the goal is to propagate the env loader into Runtime, this is not the right place to do that IMO.

**Comment by mmandlis** [comment]
> I need to understand the impetus for this PR before I can review it properly. Is there a corresponding issue?

I think this is the closest for corresponding issue https://github.com/PolymerLabs/arcs/issues/2557

File: shells/lib/env/env-base.js (hunk 2 above)
Comment (sjmiles) [reply shells/lib/env/env-base.js]: IMO, `env` and `Runtime` are solutions to the same problem. We shouldn't try to use them both.

Comment (raulverag) [reply shells/lib/env/env-base.js]: Sorry, Scott. I should have put more into the initial comment.

This entire PR a hack to unblock Maria, while I work on the correct solution. Shane asked me to look at a quick way, as well as a long-term way, to help her deal with the fact that she needs to instantiate a plan without an arc. Instantiation needs a loader and a context, both of which are currently pretty much singletons, so this PR is for pushing the loader directly into the runtime code, without attaching it to an arc. The plan for today is to do the same for the context.

File: shells/lib/env/env-base.js (hunk 3 above)
Comment (raulverag) [reply shells/lib/env/env-base.js]: Yes, but we looked at it and found the only use to be the call immediately preceding, in the constructor, which doesn't make use of the laziness. Did we miss another call?

File: shells/lib/env/env-base.js (hunk 4 above)
Comment (raulverag) [reply shells/lib/env/env-base.js]: Agreed, but the right place will take more work so this is temporary to unblock Maria, as above.

**Comment by raulverag** [comment]
After our conversation over VC, I'm deleting this PR.

**Pull request closed without merging**